qa/acceptance/Vagrantfile

```diff
@@ -0,0 +1,64 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure(2) do |config|
+
+  config.vm.define "ubuntu-1204" do |ls|
+    ls.vm.box = "elastic/ubuntu-12.04-x86_64"
+    ubuntu_common_config(ls)
+  end
+
+  config.vm.define "ubuntu-1404" do |ls|
+    ls.vm.box = "elastic/ubuntu-14.04-x86_64"
+    ubuntu_common_config(ls)
+  end
+
+  config.vm.define "centos-6" do |ls|
+    ls.vm.box = "elastic/centos-6-x86_64"
+    centos_common_config(ls)
+  end
+
+  config.vm.define "centos-7" do |ls|
+    ls.vm.box = "elastic/centos-7-x86_64"
+    centos_common_config(ls)
+  end
+
+end
+
+def centos_common_config(ls)
+  ls.vm.provider "virtualbox" do |v|
+    v.memory = 4096
+    v.cpus = 4
+  end
+ls.vm.synced_folder "../../build", "/logstash-build"
+  ls.vm.provision :shell do |sh|
+    sh.path = "sys/centos/bootstrap.sh"
+    sh.privileged = true
+  end
+
+  ls.vm.provision :shell do |sh|
+    sh.path = "sys/centos/user_bootstrap.sh"
+    sh.privileged = false
+  end
+
+  #ls.vm.network "forwarded_port", guest: 9600, host: 9601
+end
+
+def ubuntu_common_config(ls)
+  ls.vm.provider "virtualbox" do |v|
+    v.memory = 4096
+    v.cpus = 4
+  end
+  ls.vm.synced_folder "../../build", "/logstash-build"
+  ls.vm.provision :shell do |sh|
+    sh.path = "sys/ubuntu/bootstrap.sh"
+    sh.privileged = true
+  end
+
+  ls.vm.provision :shell do |sh|
+    sh.path = "sys/ubuntu/user_bootstrap.sh"
+    sh.privileged = false
+  end
+
+  #ls.vm.network "forwarded_port", guest: 9600, host: 9600
+end
```

Comment:
Did you consider a hash based configuration for the machine definition like this one?
I find it more concise easier to deal with, but It might be matter of taste.

```ruby
boxes = {
"ubuntu-1204" => { "box" => "elastic/ubuntu-12.04-x86_64" }
"centos-6" => { "box" => "elastic/centos-6-x86_64" }
}

boxes.each |machine_name, config|
machine_type = machine_name.gsub(/[^a-ZA-Z]/,'')
 config.vm.define machine_name do |ls|
     ls.vm.box = config["box"]
   ls.vm.provider "virtualbox" do |v|
     v.memory = 4096
     v.cpus = 4
   end
   ls.vm.synced_folder "../../build", "/logstash-build"
   ls.vm.provision :shell do |sh|
     sh.path = "sys/#{machine_type}/bootstrap.sh"
     sh.privileged = true
  end
 
   ls.vm.provision :shell do |sh|
     sh.path = "sys/{machine_type}/user_bootstrap.sh"
     sh.privileged = false
   end
end
```

I like the idea of letting vagrant decide the port/ips.